unit test TestQp::test_all_solvers fails after updating to scipy 1.12

**Describe the bug**
after updating from scipy 1.11.4 -> 1.12 the following unit tests started failing: -- root cause seems the same as described in https://github.com/osqp/osqp-python/issues/121#issuecomment-1925563879 (scipy.sparse.random values transposed)

```console
=================================== FAILURES ===================================
___________________________ TestQp.test_all_solvers ____________________________

self = <cvxpy.tests.test_qp_solvers.TestQp testMethod=test_all_solvers>

    def test_all_solvers(self) -> None:
        for solver in self.solvers:
            self.quad_over_lin(solver)
            self.power(solver)
            self.power_matrix(solver)
            self.square_affine(solver)
            self.quad_form(solver)
            self.affine_problem(solver)
            self.maximize_problem(solver)
            self.abs(solver)
    
            # Do we need the following functionality?
            # self.norm_2(solver)
            # self.mat_norm_2(solver)
    
            self.quad_form_coeff(solver)
            self.quad_form_bound(solver)
            self.regression_1(solver)
            self.regression_2(solver)
            self.rep_quad_form(solver)
    
            # slow tests:
            self.control(solver)
            self.sparse_system(solver)
            self.smooth_ridge(solver)
            self.huber_small(solver)
>           self.huber(solver)

cvxpy/tests/test_qp_solvers.py:105: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
cvxpy/tests/test_qp_solvers.py:370: in huber
    self.assertAlmostEqual(1.327429461061672, objective.value, places=3)
cvxpy/tests/base_test.py:39: in assertAlmostEqual
    super(BaseTest, self).assertAlmostEqual(a, b, places=places, delta=delta)
E   AssertionError: 1.327429461061672 != 0.8462713317138182 within 3 places (0.48115812934785374 difference)
----------------------------- Captured stdout call -----------------------------
The problem is QP:  True
(3, 100) (3,)
```

**Version**
 - OS: linux
 - CVXPY Version: 1.4.2

--- a/cvxpy/tests/test_qp_solvers.py
+++ b/cvxpy/tests/test_qp_solvers.py
@@ -351,10 +351,12 @@ class TestQp(BaseTest):
 
     def huber(self, solver) -> None:
         # Generate problem data
-        np.random.seed(2)
         n = 3
         m = 5
-        A = sp.random(m, n, density=0.8, format='csc')
+        data = [0.89, 0.39, 0.96, 0.34, 0.68, 0.18, 0.63 ,0.42, 0.51, 0.66, 0.43, 0.77]
+        indices = [0, 1, 2, 3, 4, 2, 3, 0, 1, 2, 3, 4]
+        indptr = [0, 5, 7, 12]
+        A = sp.csc_matrix((data, indices, indptr), shape=(m,n))
         x_true = np.random.randn(n) / np.sqrt(n)
         ind95 = (np.random.rand(m) < 0.95).astype(float)
         b = A.dot(x_true) + np.multiply(0.5*np.random.randn(m), ind95) \
@@ -367,9 +369,9 @@ class TestQp(BaseTest):
         # Solve problem with QP
         p = Problem(Minimize(objective))
         self.solve_QP(p, solver)
-        self.assertAlmostEqual(1.327429461061672, objective.value, places=3)
+        self.assertAlmostEqual(1.452797819667, objective.value, places=3)
         self.assertItemsAlmostEqual(x.value,
-                                    [-1.03751745, 0.86657204, -0.9649172],
+                                    [1.20524645, -0.85271489, -0.50838494],
                                     places=3)
 
     def equivalent_forms_1(self, solver) -> None:

